subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/ProviderCodecs.kt

```diff
@@ -144,6 +161,47 @@ class FlowProvidersCodec(
 }
 
 
+internal
+object RegisteredFlowActionCodec : Codec<RegisteredFlowAction> {
+
+    override suspend fun WriteContext.encode(value: RegisteredFlowAction) {
+        val owner = isolateOwner()
+        val flowActionClass = value.type
+        withDebugFrame({ flowActionClass.name }) {
+            writeClass(flowActionClass)
+            withFlowActionIsolate(flowActionClass, owner) {
+                write(value.parameters)
+            }
+        }
+    }
+
+    override suspend fun ReadContext.decode(): RegisteredFlowAction {
+        val flowActionClass = readClassOf<FlowAction<FlowParameters>>()
+        withFlowActionIsolate(flowActionClass, isolateOwner()) {
+            return RegisteredFlowAction(flowActionClass, read()?.uncheckedCast())
+        }
+    }
+
+    private
+    inline fun <T : MutableIsolateContext, R> T.withFlowActionIsolate(flowActionClass: Class<*>, owner: IsolateOwner.OwnerFlowScope, block: T.() -> R): R {
+        withIsolate(IsolateOwner.OwnerFlowAction(flowActionClass, owner)) {
+            withPropertyTrace(PropertyTrace.BuildLogicClass(flowActionClass.name)) {
+                return block()
+            }
+        }
+    }
+
+    private
+    fun IsolateContext.isolateOwner(): IsolateOwner.OwnerFlowScope {
```

Comment:
nit: maybe `verifiedIsolateOwner` would stand out a bit more at the call sites. But the class is small enough for it not to matter much.